### DISCUSSION TITLE
Updated guide for Chakra Setup with Remix + React 18?

### DISCUSSION THREAD

**Discussion by tyteen4a03** [chakra-ui/chakra-ui#6480] [Q&A]
I'm trying to set up a new Chakra project with React 18. The guide at https://chakra-ui.com/getting-started/remix-guide seems like it's for React 17 (as it doesn't use renderToPipeableStream). Is there an updated version of the guide available?

**Comment by bmaycc** [comment]
Chakra works fine with Remix, React 18 and renderToString.  The renderToPipeableStream issue is a known concern of EmotionJS, which has a thread about it here: https://github.com/emotion-js/emotion/issues/2800.

**Comment by WesleyYue** [comment]
Remix has a working example for streaming with Chakra [here](https://github.com/remix-run/examples/tree/main/chakra-ui)

**Reply by bmaycc** [reply]
Thanks a lot for pointing this out @WesleyYue ! - I tried it out with my app, and it does actually work for the most part, using Remix's defer and all!  I kept getting lost in the threads out there about how much Emotion needs more work, even though this renderToNodeStream function has existed for a while...

One caveat:  I'm getting the error "This Suspense boundary received an update before it finished hydrating. This caused the boundary to switch to client rendering. The usual way to fix this is to wrap the original update in startTransition."  This doesn't SEEM to affect the behavior, and there are several threads about this warning in the Remix discussions and such, seems there's some things still need sorting out. I'd be curious if anyone's running this wihtout that issue.

Not sure I'm rushing to use it *yet* in my prod app, but really great to see this working as expected :D

Thanks again!

**Reply by WesleyYue** [reply]
Yeah, hydration is really fragile. I haven't been able to fully get it working confident enough to use defer in production.
See:
- https://github.com/remix-run/remix/issues/5165#issuecomment-1437777284
- https://github.com/remix-run/remix/discussions/5244#discussioncomment-5029201

**Reply by nihalanisumit** [reply]
Yeah, even I am unable to get defer working!
Works locally but not when deployed to vercel.
Any idea what might be the case? @WesleyYue

**Reply by muco-rolle** [reply]
Has anyone tried Remix Island (https://github.com/Xiphe/remix-island)? It's working well for me, even though I need to explore more.
